lib/pages/home/wallet/wallet.dart

```diff
@@ -43,7 +48,7 @@ class WalletPage extends StatelessWidget {
                   await Future.delayed(Duration(milliseconds: 200), () {});
                 },
                 child: CompanionListView(
-                  children: state.currencies.map((c) => _buildCurrencyRow(context, c)).toList(),
+                  children: _currencies.asMap().entries.map((entry) => _buildCurrencyRow(context, entry.key, entry.value)).toList(),
```

Comment:
Looks better, but what i meant was using the List.generate here to get both the index and the currency. For example:  
```
children: List.generate(state.currencies.length, (index) => _buildCurrencyRow(context, index, state.currencies[index]))
```

You can then also remove the three lines at 39 to 41.

Reply:
Ah, that's clever. I'm learning as I'm going :)